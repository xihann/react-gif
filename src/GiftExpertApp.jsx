import { useState } from "react"
import {AddCategory, GifGrid} from "./components";


export const GiftExpertApp = () => {

  const [categories, setCategories] = useState(['Dragon Ball']);

  const onAddCategory = (newCategory) => {

    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>Gift Search App</h1>
      <p>Solo se mostrar√°n 10 GIFs</p>
      {/* Input */}
      <AddCategory
        onNewCategory={value => onAddCategory(value)}
      />

      {/* Listado de Gift */}
      {
        categories.map(category => (
          <GifGrid
            key={category}
            category={category}
          />
        ))
      }
    </>
  )
}

export default GiftExpertApp
