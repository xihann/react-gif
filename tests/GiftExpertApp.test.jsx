import { act, render, screen } from "@testing-library/react";
import GiftExpertApp from "../src/GiftExpertApp";

describe('Pruebas en  <GiftExpertApp/>', () => {
    test('deberia hacer match con el snapshot de <GiftExpertApp/> ', () => {

        const { container } = render(< GiftExpertApp />);

        expect(container).toMatchSnapshot();
    })

    test('no debe agregar una categoria duplicada ', () => {

        render(< GiftExpertApp />);
       

        // Simulamos la adición de una nueva categoría
        act(() => {
            screen.getByText('Pokémon').click();
        });

        // Verificamos que solo hay una instancia de 'Pokémon'
        const categories = screen.getAllByText(/Pokémon/i);
        expect(categories).toHaveLength(1); // Solo debe haber una categoría Pokémon
        expect(screen.getByText('Pokémon')).toBeTruthy();

    })

})
