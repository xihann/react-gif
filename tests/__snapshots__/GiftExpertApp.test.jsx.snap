// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GiftExpertApp/> deberia hacer match con el snapshot de <GiftExpertApp/>  1`] = `
<div>
  <h1>
    Gift Toshi App
  </h1>
  <p>
    Solo se mostraran 4 gifs
  </p>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gift"
      type="text"
      value=""
    />
  </form>
  <h3>
    Pok√©mon
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
